configs/config.go

```diff
@@ -0,0 +1,52 @@
+package configs
+
+import (
+	"io/ioutil"
+	"log"
+	"time"
+
+	yaml "gopkg.in/yaml.v3"
+)
+
+const ConfigurationPath = "gz-beat-shipper-conf.yml"
+
+type Configuration struct {
+	Host     string `yaml:"host"`
+	Port     string `yaml:"port"`
+	Path     string `yaml:"path"`
+	Registry string `yaml:"registry"`
+	Recheck  string `yaml:"recheck"`
+}
+
+func (c *Configuration) GetConfiguration() *Configuration {
+	yamlFile, err := ioutil.ReadFile(ConfigurationPath)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = yaml.Unmarshal(yamlFile, c)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return c
+}
+
+func (c *Configuration) GetRecheckInSeconds() int32 {
+	t, err := time.ParseDuration(c.Recheck)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	s := t.Seconds()
```

Comment:
What value would `t` have if an error is found?

Reply:
If an error is found the code will exist in the `log.Fatal` part. If this code is executed successfully `time.ParseDuration(c.Recheck)` then this conversion can be done `t.Seconds()` and that's why it doesn't return error in the definition `func (d Duration) Seconds() float64 {`